astropy/timeseries/tests/test_downsample.py

```diff
@@ -6,12 +6,15 @@
 
 from astropy import units as u
 from astropy.time import Time
+from astropy.utils.exceptions import AstropyUserWarning
 
 from astropy.timeseries.sampled import TimeSeries
 from astropy.timeseries.downsample import aggregate_downsample, reduceat
 
+
 INPUT_TIME = Time(['2016-03-22T12:30:31', '2016-03-22T12:30:32',
-                   '2016-03-22T12:30:33', '2016-03-22T12:30:34'])
+                   '2016-03-22T12:30:33', '2016-03-22T12:30:34',
+                   '2016-03-22T12:30:35'])
```

Comment:
```suggestion
                   '2016-03-22T12:30:35']).ns
# Avoid precision problems with floating-point comparisons on 32bit
if np.finfo(float).dtype == np.float32:
    INPUT_TIME += 1 * u.ns
```
maybe try this instead...

Reply:
I tried both `INPUT_TIME += 1 * u.ns` and `INPUT_TIME[1:] += 1 * u.ns` and they don't work.
I'll see if subtracting works. Maybe the code for getting precedence of start times is causing extra precision errors?

Reply:
Good catch with `INPUT_TIME[1:]` – I had overlooked that this is testing the `time_bin_size` case, so incrementing `INPUT_TIME[0]` would shift all start values as well...
Maybe instead of the above just define something like
```python
    if np.finfo(float).dtype == np.float32:
        time_bin_incr = (1 - 1e-6) * u.s
    else:
        time_bin_incr = 1 * u.s
```
and then proceed with 1 *, 2 * `time_bin_incr` etc.
Lastly, I'd be entirely fine with marking anything that still would not pass `skipif` or `xfail` for 32bit.
Just noticed a comment about `Time` in second representation not fitting into 32bit integers; no idea if this might be in any way relevant here, just there might be issues beyond simple floating point precision...